jaeger-core/src/main/java/com/uber/jaeger/Tracer.java

```diff
@@ -272,13 +284,23 @@ boolean isRPCServer() {
 
     @Override
     public io.opentracing.Span start() {
+      // find preferredParent, it is first childOf
+      Reference preferredParent = references.isEmpty() ? null : references.get(0);
+      for (Reference reference: references) {
+        if (References.CHILD_OF.equals(reference.getType()) &&
+                References.FOLLOWS_FROM.equals(preferredParent.getType())) {
+          preferredParent = reference;
+          break;
+        }
+      }
+
       SpanContext context;
-      if (parent == null) {
+      if (preferredParent == null) {
         context = createNewContext(null);
-      } else if (parent.isDebugIDContainerOnly()) {
-        context = createNewContext(parent.getDebugID());
+      } else if (preferredParent.referencedContext().isDebugIDContainerOnly()) {
```

Comment:
No, it's fine, just a note to myself. We want to be able to parse baggage from the request even when there's no trace context. We do it in Go client, but not in Java. I will do a separate pr for that.